### PR TITLE
Source-DOMO migration

### PR DESCRIPTION
**Pull request by CoopTang**
## What
* Adds `source-domo` to Airbyte integrations. This is a transfer from the original `tap-domo` written earlier this year

## How
Rewrites the singer tap to be used with the Airbyte CDK

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

## Notes
- This requires refactoring before it can be considered for Airbyte since this is very client specific.
- The config is required for tests. The config I used can be found within the 1Pass vault for Purple Lotus.

### CODE REVIEW

File: airbyte-integrations/connectors/source-domo/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                "read",
+                "--config",
+                "./secrets/config.json",
+                "--catalog",
+                "./catalogs/configured_catalog.json"
+            ],
+            "justMyCode": false
+        }
+    ]
+}
```
Comment (alerrian) [inline airbyte-integrations/connectors/source-domo/.vscode/launch.json:49]: I think this file accidentally made it in to the repo. I saw the `.vscode` got added to `.gitignore` but this one made it in anyway.

File: airbyte-integrations/connectors/source-domo/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "python.formatting.provider": "black",
+    "python.terminal.activateEnvironment": true
+}
```
Comment (alerrian) [inline airbyte-integrations/connectors/source-domo/.vscode/settings.json:4]: Same as the file above.

File: airbyte-integrations/connectors/source-domo/source_domo/source.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    def streams(self, config: Mapping[str, Any]) -> List[DOMOStream]:
+        client_id = config["client_id"]
+        client_secret = config["client_secret"]
+        data_specs = config["data_specs"]
+
+        client = DOMOClient(client_id=client_id, client_secret=client_secret)
+        streams = []
+        for name, data_spec in data_specs.items():
+            streams.append(DOMOStream(client, data_spec))
+        return streams
```
Comment (alerrian) [inline airbyte-integrations/connectors/source-domo/source_domo/source.py:79]: Is this dynamically creating/generating streams for each dataset in the incoming `data_specs` object? If so, we need to test that this is actually doing what it is supposed to in the Airbyte UI. Specifically, auth and schema reading/generation.

Comment (CoopTang) [reply airbyte-integrations/connectors/source-domo/source_domo/source.py]: It was. Refactored to hardcoded streams.

**Review by alerrian — APPROVED** [review]
This looks great. Thanks for taking the time to refactor the stream file when we asked about it.

**Pull request merged**